### ISSUE TITLE
Biomedisa + Slicer extension works with CUDA 12.2

### ISSUE THREAD

**Issue by mcranium** [biomedisa/biomedisa#10]
This is just a comment that Biomedisa, including the new Slicer Plugin works with CUDA version 12.2 as well. I think this should be reflected in the README, as it is important for those wanting to set up Biomedisa locally now that the Slicer extension is out (will be an absolute game changer for many). I used a box-standard Ubuntu 24.04 installation and the 535.183.01 driver for my 3060 Ti  card. Note that the location of the CUDA executables is under `/usr/lib/cuda` instead of `/usr/local/cuda-11.8` (needs to be adjusted when configuring). Let me know if you want me to do some specific tests with my configuration.

**Comment by philipp-loesel** [comment]
**Smart Interpolation installation**
Biomedisa's Smart Interpolation is compatible with any CUDA version. However, users must ensure that the CUDA version matches the CUDA compute capabilities of their NVIDIA GPU architecture and that the NVIDIA driver version meets the requirements: [NVIDIA Documentation](https://docs.nvidia.com/deeplearning/cudnn/latest/reference/support-matrix.html). For older CUDA versions, this link is also helpful: [Matching SM Architectures](https://arnon.dk/matching-sm-architectures-arch-and-gencode-for-various-nvidia-cards/).

**TensorFlow installation**
The Deep Learning module is more complex. On Ubuntu, we currently support only CUDA 11.8 + cuDNN 8.8.0 + TensorFlow 2.13, though other combinations may also work. To simplify installation instructions, we have specified only CUDA 11.8, even if TensorFlow installation is skipped.

**Further steps**
I noted in the instructions that any CUDA version can be used if TensorFlow installation is skipped. If you can document your installation steps, we can include that in the instructions for Ubuntu 24.04. Did you install CUDA using `sudo apt install nvidia-cuda-toolkit`?

**Comment by mcranium** [comment]
I am sorry for the late reply. Actually, after an update (probably), I ran into troubles getting the Slicer extension to run. Biomedisa itself indeed seems to run with various CUDA versions. Now I have the situation that despite that Biomedisa works, for some reason, I am not able anymore to install pycuda into `PythonSlicer` (`Failed to build installable wheels for some pyproject.toml based projects (pycuda)`. Also mpi4py did not install as expected, but I could get around it via [this](https://github.com/mpi4py/mpi4py-publish). I am currently on CUDA 12.5 and Nvidia driver 555 (proprietary), but willing to switch, if it means getting the Slicer extension to run. Any help is greatly appreciated.

**Comment by biomedisa** [comment]
Could you clarify what you mean by "Biomedisa works". Does that mean you installed it according the [command-line based](https://github.com/biomedisa/biomedisa/blob/master/README/ubuntu2204_cuda11.8_gpu_cli.md) instructions adapted to your CUDA version?

The error suggests that the workaround for PyCUDA and mpi4py failed. While pre-built Python wheels are available, I recommend building PyCUDA from source to match your specific system architecture. Note that 3D Slicer uses `python3.9`. To create wheels for both packages, ensure you have `python3.9-dev` installed. Since Ubuntu 22.04 and 24.04 uses `python>=3.10`, you'll need to install `python3.9` alongside the existing Python installation.
```
sudo add-apt-repository ppa:deadsnakes/ppa
sudo apt-get install python3.9 python3.9-dev  
sudo apt install python3.9-distutils
```
Ensure the following path exists: `/usr/include/python3.9`.

Change to your Slicer environment, for example:
```
cd Slicer-5.7.0-2024-07-20-linux-amd64/bin
```
Remove existing installations:
```
python3.9 -m pip uninstall mpi4py pycuda
./PythonSlicer -m pip uninstall mpi4py pycuda
```
Now, you have two options:

**Option 1:**
Follow the [istallation instructions](https://github.com/biomedisa/biomedisa/blob/master/README/ubuntu2204_cuda11.8_gpu_slicer.md#nasty-double-installation-of-pip-packages-hopefully-will-be-fixed-soon) precisely, installing PyCUDA and mpi4py using `python3.9` first and then using `./PythonSlicer`. This will build wheels that can also be used by `./PythonSlicer`.

**Option 2:**
Directly install them in 3D Slicer environment, including the required paths. For mpi4py:
```
./PythonSlicer -c "import os,sys; os.environ['C_INCLUDE_PATH'] = '/usr/include/python3.9'; os.system(f'{sys.executable} -m pip install --no-cache-dir mpi4py')"
```

For PyCUDA:
```
./PythonSlicer -c "import os,sys; os.environ['C_INCLUDE_PATH'] = '/usr/include/python3.9'; os.environ['CPLUS_INCLUDE_PATH'] = '/usr/include/python3.9'; os.environ['CC'] = '/usr/bin/gcc'; os.environ['CXX'] = '/usr/bin/g++'; os.system(f'{sys.executable} -m pip install --no-cache-dir pycuda')"
```
**Note:**
The `--no-cache-dir` option is added to avoid potential conflicts with prebuild packages.

**Comment by mcranium** [comment]
I am sorry, with "biomedisa works", I meant that I could process files using the smart interpolation function from the command line (yes, I followed the instructions for the command line based installation).

Thank you very much! Your further instructions were very helpful and I managed to get the smart interpolation in the Slicer extension to run. I have to admit that I do not fully understand what the issue with the Python wheels was. But after using your option 2 for installing `mpi4py` and `pycuda` into `PythonSlicer`, I had no issues anymore. Maybe, these instructions could be used to provide more distribution independent installation instructions.
